tests/test_workflows.py

```diff
@@ -119,4 +119,39 @@ async def event_workflow(c: Context):
         assert s.w_output == "Hello from outside! B A"
 
 
-# TODO: test concurrent
+current_time = None
+rand_value = None
+unique_id = None
+
+
+@async_test
+async def test_utils_methods(t: WorkflowTest):
+    def is_durable(a: any, b: any):
+        if a != b:
+            raise Exception(f"not durable: {a} != {b}")
+        pass
+
+    @workflow()
+    async def utils_workflow(context: Context):
+        global current_time, rand_value, unique_id
+        a = await context.utils.now()
+        b = await context.utils.random(1, 10)
+        c = await context.utils.uuid4()
+        if current_time is None and rand_value is None and unique_id is None:
+            current_time = a
+            rand_value = b
+            unique_id = c
+        else:
+            is_durable(a, current_time)
+            is_durable(b, rand_value)
+            is_durable(c, unique_id)
+        context.sleep(timedelta(1))
+        return str(a), b, str(c)
+
+    backends = [
+        FSBackend("./logs"),
+        RedisBackend(host="localhost", port=6380, password="password"),
+    ]
+
+    for backend in backends:
+        await t.step(backend).exec_workflow(utils_workflow)
```

Comment:
can you check the type of the return or even more precisely (for random between 1 and 10) ?

Reply:
yeah!